src/store-enhancer.js

```diff
@@ -22,37 +25,69 @@ type StoreEnhancerArgs = {
   history: History,
   location: Location,
   createMatcher?: Function,
-  passRouterStateToReducer?: bool
+  passRouterStateToReducer?: bool,
+  immutable?: bool
 };
 
 export default ({
   routes: nestedRoutes,
   history,
   location,
   createMatcher = matcherFactory,
-  passRouterStateToReducer = false
+  passRouterStateToReducer = false,
+  immutable = false
 }: StoreEnhancerArgs) => {
   validateRoutes(nestedRoutes);
   const routes = flattenRoutes(nestedRoutes);
+  let assign = _assign;
+  let omit = _omit;
+  let get = (obj, prop) => obj[prop];
+  if (immutable === true) {
+    assign = (immutableObj, obj2) => {
+      if (typeof immutableObj === 'undefined') {
+        return immutableObj;
+      }
+      // don't want router object to be  deeply converted via Immutable.fromJS
+      // https://facebook.github.io/immutable-js/docs/#/Map/merge
```

Comment:
I'm curious why you don't want the router object to be deeply converted. It seems to make more consistency if the store is immutable to convert it